### ISSUE TITLE
Firebase component/ localstorage/indexeddb/webrtc data/physics/networking similar projects

### ISSUE THREAD

**Issue by runvnc** [ngokevin/kframe#12]
TLDR: is there something like this but using localStorage/indexedDB and webrtc data channels?

Hello, low priority question since I'm sure you are extremely busy.  Thank you for all of your amazing work on a-frame.

To me, a-frame seems to be the leader in terms of the technical foundation for the possibility of having a truly open metaverse.  In that I mean building open systems from a-frame components that  eventually compete to some degree with things like High Fidelity/Project Sansar or maybe Garry's Mod etc. in terms of social/user-built VR.  I understand the project goals are not the same, its more like a-frame is a starting point for programmers.

This component, aframe-firebase-component, would play a key role in someone trying to recreate that type of shared VR experience or world.  Along with a physics system.  And some way to synchronize the physics and updates in general over clients.  Which is the hardest part of all of it.

It makes practical sense to use Firebase. But in the ideal case, we would be able to use local storage and webRTC data channels to enable the same type of thing without relying on a central server/commercial service.  Basically, the goal would be something similar to a High Fidelity domain (which is a server the uses UDP and handles syncing the VR environment/world that a specific user owns) but it runs entirely in a browser process (maybe SharedWorker etc.).

Do you know of any projects along those lines?  Or are you basically having to do all of that yourself and so we are on making Firebase do basic stuff still. Thanks.

I am wondering if people are not even really seriously trying to do that inside of browsers and instead essentially building their own optimized browsers/3d graphics+browser bundles just on the basis of the assumption that they can't optimize across arbitrary browsers.

**Comment by ngokevin** [comment]
I definitely see A-Frame powering open social experiences such as HF or Garry's Mod one day. 

Synchronization is tough, some point we'll have to put resources into enabling tools for multiplayer. I don't know any projects yet, would love to see others try out WebRTC.

**Comment by runvnc** [comment]
OK, thanks for answering :). I experimented with three.js and webrtc data channels a few years ago but as you said, synchronization is tough (I would say super tough).  Maybe at some point I can play around with webrtc in relation to a-frame.  Closing for now.  Feel free to re-open or create another issue like this if you think it needs an open issue on github.
